packages/ember-data/lib/adapters/rest_adapter.js

```diff
@@ -171,6 +171,58 @@ export default Adapter.extend({
   defaultSerializer: '-rest',
 
   /**
+    By default, the RESTAdapter will send the query params sorted alphabetically to the
+    server.
+
+    For example:
+    ```js
+      store.find('posts', {sort: 'price', category: 'pets'});
+    ```
+    will generate a requests like this `/posts?category=pets&sort=price`, even if the
+    parameters were specified in a different order.
+
+    That way the generated URL will be deterministic and that simplifies caching mechanisms
+    in the backend.
+
+    Setting `sortQueryParams` to a falsey value will respect the original order.
+
+    In case you want to sort the query parameters with a different criteria, set
+    `sortQueryParams` to your custom sort function.
+
+    ```js
+    export default DS.RESTAdapter.extend({
+      sortQueryParams: function(params) {
+        var sortedKeys = Object.keys(params).sort().reverse();
+        var len = sortedKeys.length, newParams = {};
+
+        for (var i = 0; i < len; i++) {
+          newParams[sortedKeys[i]] = params[sortedKeys[i]];
+        }
+        return newParams;
+      }
+    });
+    ```
+
+    @method sortQueryParams
+    @param {Object} obj
+    @return {Object}
```

Comment:
I don't think the order of iteration over keys in an objects is guaranteed to always be the order that the keys were assigned to the object. http://www.ecma-international.org/ecma-262/5.1/#sec-12.6.4 Although in practice it sounds like all browsers follow that pattern so maybe this isn't an issue. https://stackoverflow.com/questions/280713/elements-order-in-a-for-in-loop